### PR TITLE
Update Workflow options

### LINKED ISSUE #55: Update Workflow nodes with new options
Tower workflow nodes have options that are not included in our role, like inventory and scm, we need to add those in for use.

### PR DESCRIPTION
**Pull request by sean-m-sullivan**
### What does this PR do?
Adds missing module options from workflow templates and workflow nodes

### How should this be tested?
Automated tests

### Is there a relevant Issue open for this?
resolves #55

### Other Relevant info, PRs, etc.
Requires PR to update simiplified playbooks.
#56

### CODE REVIEW

**Review by Tompage1994 — APPROVED** [review]
Looks good.

@sean-m-sullivan All these new options applied against the workflow module, are they new to the awx collection module or were they just missed in the first place? If they were missed does that mean we need to make a check against all the other modules to make sure we hit all options?

**Comment by sean-m-sullivan** [comment]
@Tompage1994 I didn't need to update the Readme, these were all in there when I wrote the module, but somehow missed them. We should check the other modules as well. Wish there was a easy way to do that, but I was using the role as a reference when someone asked about these options and noticed them missing.

**Comment by sean-m-sullivan** [comment]
Rebasing this and then shall be cleaner

File: playbooks/tower_configs_export_model/tower_workflows.yml [hunk 1]
```diff
@@ -12,23 +12,17 @@ tower_workflows:
...
     ask_limit_on_launch: false
-    webhook_service: ''
-    webhook_credential:
     organization:
       name: Default
       type: organization
     related:
       schedules: []
       workflow_nodes:
-        - extra_data: {}
```
Comment (kedark3) [inline playbooks/tower_configs_export_model/tower_workflows.yml:23]: We can no longer pass this `extra_data`?

File: roles/workflow_job_templates/tasks/add_workflows_schema.yml [hunk 2]
```diff
@@ -3,6 +3,16 @@
 - name : Create the Workfow Node
   awx.awx.tower_workflow_job_template_node:
     identifier :                    "{{ workflow_loop_node.identifier }}"   # internal identification ID
+    extra_data:                     "{{ workflow_loop_node.extra_data | default (omit) }}"
```
Comment (kedark3) [inline roles/workflow_job_templates/tasks/add_workflows_schema.yml:6]: Can we standardize between extra_vars and extra_data? I like former better. Even though modules aren't accepting same variable, we can do maybe:

```
extra_data:                     "{{ workflow_loop_node.extra_vars | default (omit) }}"
```

File: roles/workflow_job_templates/tasks/add_workflows_schema.yml [hunk 3]
```diff
@@ -3,6 +3,16 @@
...
+    inventory:                      "{{ workflow_loop_node.inventory | default (omit) }}"
+    scm_branch:                     "{{ workflow_loop_node.scm_branch | default (omit) }}"
+    job_type:                       "{{ workflow_loop_node.job_type | default (omit) }}"
+    job_tags:                       "{{ workflow_loop_node.job_tags | default (omit) }}"
+    skip_tags:                      "{{ workflow_loop_node.skip_tags | default (omit) }}"
+    limit:                          "{{ workflow_loop_node.limit | default (omit) }}"
+    diff_mode:                      "{{ workflow_loop_node.diff_mode | default (false) }}"
+    verbosity:                      "{{ workflow_loop_node.verbosity | default (omit) }}"
+    workflow_job_template:          "{{ workflow_loop_node.workflow_job_template.name | default( workflow_loop_node.workflow_job_template | default (omit) ) }}"
```
Comment (kedark3) [inline roles/workflow_job_templates/tasks/add_workflows_schema.yml:15]: Does double default like this work? Have you tested it? Just curious.

**Review by kedark3 — COMMENTED** [review]
Couple of comments. If you could please respond, we should be good to go. And I can Approve/Merge it.

File: roles/workflow_job_templates/tasks/add_workflows_schema.yml (hunk 3 above)
Comment (sean-m-sullivan) [reply roles/workflow_job_templates/tasks/add_workflows_schema.yml:15]: Yes it was working in the last version, and works with both data models in the testing. It was thinkings like Limit, Diff mode, and other options that were missing, and order got changed in this go around.

File: roles/workflow_job_templates/tasks/add_workflows_schema.yml (hunk 2 above)
Comment (sean-m-sullivan) [reply roles/workflow_job_templates/tasks/add_workflows_schema.yml:6]: So I went off the module here [extra data in workflow nodes](https://github.com/ansible/awx/blob/devel/awx_collection/plugins/modules/tower_workflow_job_template_node.py) Where on the node its classfied in Ansible tower as extra data for things that are passed at runtime. 

So in this case a Workflow node passes extra data Along with the Job templates original extra vars. and combines them to one item that is consumed. 

Extra data is only passed when Prompt on box is checked in extra vars, on a job template, and when specified when creating that workflow. Its in addition to the workflow extra vars as well.

**Review by aoyawale — APPROVED** [review]

**Review by aoyawale — APPROVED** [review]

**Comment by aoyawale** [comment]
approved

File: playbooks/tower_configs_export_model/tower_workflows.yml (hunk 1 above)
Comment (sean-m-sullivan) [reply playbooks/tower_configs_export_model/tower_workflows.yml:23]: It raises hell when its included and the template hasn't been set to prompt for extra vars.

File: roles/workflow_job_templates/tasks/add_workflows_schema.yml (hunk 2 above)
Comment (sean-m-sullivan) [reply roles/workflow_job_templates/tasks/add_workflows_schema.yml:6]: Got a screenshot of the underlying documentation, so including things here, its extra vars but its called extra data in the api. 
extra_data1
extra_data_api

Comment (kedark3) [reply roles/workflow_job_templates/tasks/add_workflows_schema.yml:6]: I think the difference in my opinion is `extra_vars` is for WF and `extra_data` is node level. Is that right summary? Maybe that's something you can add into README ? :)

Comment (sean-m-sullivan) [reply roles/workflow_job_templates/tasks/add_workflows_schema.yml:6]: yes. Will add something here soon.

Comment (sean-m-sullivan) [reply roles/workflow_job_templates/tasks/add_workflows_schema.yml:6]: Updated Readme

**Pull request merged**